src/libcore/task/wake.rs

```diff
@@ -120,7 +133,16 @@ impl LocalWaker {
     /// on the current thread.
     #[inline]
     pub unsafe fn new(inner: NonNull<dyn UnsafeWake>) -> Self {
-        LocalWaker { inner }
+        LocalWaker(Waker::new(inner))
+    }
+
+    /// Borrows this `LocalWaker` as a `Waker`.
+    ///
+    /// `Waker` is nearly identical to `LocalWaker`, but is threadsafe
+    /// (implements `Send` and `Sync`).
+    #[inline]
+    pub fn as_waker(&self) -> &Waker {
```

Comment:
Is this actually safe? With scoped threads it seems like you could use this to pass a `LocalWaker` to another thread as a `&Waker`?

Reply:
Yep, you can! And that's totally safe, because you can only call `.wake()` on the inner `UnsafeWake` in a `Wake`-- you can't call `.wake_local()`.